src/pyscipopt/scip.pyx

```diff
@@ -557,6 +557,25 @@ cdef class Constraint:
         constype = bytes(SCIPconshdlrGetName(SCIPconsGetHdlr(self.cons))).decode('UTF-8')
         return constype == 'quadratic'
 
+cdef class BendersDecomp:
```

Comment:
Why not call it Benders? This would make it consistent with the C naming.

Reply:
And there should be a doc string for this class

Reply:
The reason that it is not called `Benders`, is that the class wrapping around the C plugin type is called `Benders` (in benders.pxi). So there would be a clash with the naming. I am happy for a different name, that was the best that I could come up with.

Reply:
I see. So there is a generic benders `BendersDecomp` already included in SCIP and people should have the possibility to write their own custom benders decomposition based on the plugin type `Benders`? What about calling `BendersDecomp` `BendersDefault` then?

Reply:
Your comment just made me see a limitation of my approach. I created `BendersDecomp` to allow users to easily have access to public Benders functions, such as `updateLowerbounds`. Now, if the user creates their own `Benders` plugin, then they won't have access to these functions.

Would something like this work? The function `updateLowerbounds` is changed to `updateBendersLowerbounds` and in a member function of `Model`. The arguments of `updateBendersLowerbounds` is a dictionary `lowerbounds` and the `name` of the Benders' decomposition. `name` defaults to `default`, which then means that the default Benders' decomposition is called. In `updateBendersLowerbounds`, the C struct `SCIP_BENDERS` is retrieved by calling `SCIPfindBenders`.

Does that sounds reasonable?

Reply:
So you wouldn't need a new class `BendersDecomp` at all, right? Yes, this sounds very reasonable. Still, I find it unintuitive to pass the *name* of my custom benders when I have the instance of the class readily available.

Reply:
Yes, we would remove `BendersDecomp`.

It is a little unintuitive. Do have an idea on how to pass the `SCIP_BENDERS` pointer to the C function if we are using the default Benders' decomposition? If we have a custom Benders' decomposition python object, then we could request that as an input. But the default Benders' decomposition plugin doesn't have a python object.

Reply:
Can't we generate a PyBenders from the default benders whenever the user does not specify his own custom benders?

Reply:
Wait, do we even need to generate a PyBenders? isn't is enough to have the C pointer from `findBenders()` lying around? So either you ask for the C pointer of the default benders or you take the pointer that comes with your custom benders instance.

Reply:
That makes sense. So in `updateBendersLowerbounds`, we take a `Benders` object as a parameter, which defaults to None. If `Benders == None`, then we just use the default Benders' decomposition, otherwise we use the `SCIP_BENDERS` pointer from `Benders`.

Good idea.

Reply:
I have made some changes. I was unsure about setting the default parameter of `benders` to `None` in `updateBendersLowerbounds`.

Reply:
Looks good to me. Excpet you might want to remove the spaces around the asignment in the parameters list. It seems, this is against PEP8 style.

Reply:
How is this?

Reply:
Awesome!